ksonnet-util/legacy-types.libsonnet

```diff
@@ -15,12 +15,12 @@
   },
   rbac+: {
     v1+: {
-      policyRule:: $.rbac.v1beta1.clusterRole.rulesType,
-      subject:: $.rbac.v1beta1.clusterRoleBinding.subjectsType,
+      policyRule:: $.rbac.v1.clusterRole.rulesType,
+      subject:: $.rbac.v1.clusterRoleBinding.subjectsType,
     },
     v1beta1+: {
-      policyRule:: $.rbac.v1beta1.clusterRole.rulesType,
-      subject:: $.rbac.v1beta1.clusterRoleBinding.subjectsType,
+      policyRule:: $.rbac.v1.clusterRole.rulesType,
+      subject:: $.rbac.v1.clusterRoleBinding.subjectsType,
```

Comment:
Given that this refers to v1beta1, should this remain unchanged? If you are using `k.rbac.v1beta1` you could migrate to `k.rbac.v1` instead (which you are already patching).

Reply:
The expected end result here is that the users of these libraries should be able to easily migrate to to Kubernetes API 1.22+, which doesn't expose the v1beta1 API at all.  With this in mind, I see these approaches:
- since the changes to RBAC were minimal according to the documentation, we could just start using the `v1` API instead of `v1beta1` directly (implemented in this PR).
- since the v1beta1 API won't be available anymore in 1.22+, we could remove the `v1beta1` key of the RBAC object altogether, making it fail early if something still uses the deprecated API.

Reply:
> since the changes to RBAC were minimal according to the documentation, we could just start using the v1 API instead of v1beta1 directly (implemented in this PR).

I would prefer to go with this option first.

Reply:
@jvrplmlmn okay, let me add a TODO note about removing the `v1beta1` line at some point in the future.

Reply:
Don; let me know if this PR looks sufficiently okay to give it a green light :-)